nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-service-aws/src/main/java/org/apache/nifi/kafka/service/aws/AmazonMSKConnectionService.java

```diff
@@ -42,6 +73,56 @@ public class AmazonMSKConnectionService extends Kafka3ConnectionService {
             .defaultValue(SaslMechanism.AWS_MSK_IAM)
             .build();
 
+    public static final PropertyDescriptor AWS_WEB_IDENTITY_TOKEN_PROVIDER = KafkaClientComponent.AWS_WEB_IDENTITY_TOKEN_PROVIDER;
+
+    private static final int MIN_SESSION_DURATION_SECONDS = 900;
+    private static final int MAX_SESSION_DURATION_SECONDS = 3600;
+
+    public static final PropertyDescriptor AWS_WEB_IDENTITY_SESSION_TIME = new PropertyDescriptor.Builder()
+            .name("AWS Web Identity Session Time")
+            .description("Session time in seconds for AWS STS AssumeRoleWithWebIdentity (between 900 and 3600 seconds).")
+            .dependsOn(
+                    KafkaClientComponent.AWS_ROLE_SOURCE,
+                    AwsRoleSource.WEB_IDENTITY_TOKEN
+            )
+            .required(false)
+            .defaultValue(String.valueOf(MAX_SESSION_DURATION_SECONDS))
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor AWS_WEB_IDENTITY_STS_REGION = new PropertyDescriptor.Builder()
+            .name("AWS Web Identity STS Region")
+            .description("Region identifier used for the AWS Security Token Service when exchanging Web Identity tokens.")
+            .dependsOn(
+                    KafkaClientComponent.AWS_ROLE_SOURCE,
+                    AwsRoleSource.WEB_IDENTITY_TOKEN
+            )
+            .required(false)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor AWS_WEB_IDENTITY_STS_ENDPOINT = new PropertyDescriptor.Builder()
+            .name("AWS Web Identity STS Endpoint Override")
```

Comment:
Recommend removing `Override`.
```suggestion
            .name("AWS Web Identity STS Endpoint")
```